src/theme/material/templates/plugin-column-filter-date-between.tpl.html

```diff
@@ -0,0 +1,34 @@
+<ng-template key="plugin-column-filter-date-between.tpl.html"
+             let-$columnFilter
+             let-$plugin="plugin"
+             let-$vscroll="vscroll">
+
+	<mat-form-field class="q-grid-column-filter-input half"
+	                *q-grid-date="let isValid='isValid'"
+	                [ngClass]="{'mat-form-field-invalid': !isValid(inputLeft.value)}">
+		<input q-grid-focus
+		       matInput
+		       #inputLeft
+		       placeholder="From"
+		       [matDatepicker]="leftPicker"
+		       [ngModel]="$columnFilter.expression.right[0] | qGridDate:'yyyy-MM-dd'"
```

Comment:
can we have here $columnFilter.value? not expression. expression should be encapsulated from UI. value can be string | number | date | array, why not?

Reply:
that's would be fine if value can be string | number | date. But array type (exactly in case of 'between' operator) makes me slightly confuse. 
I this case we should have some kind of getter/setter logic for `value` property to switch its type between array and primitive. Because in _default-between.tpl_ and _date-between.tpl_ UI we have two fields, the first should refer ngModel to `blahblah[0]` and the last should refer to `blahblah[1]`. And it will raised an error if `value` was previously defined as number or Date, for example.
I can implement both getter and setter for `$columnFilter.value`. 
Or we can have `$columnFilter.left` and `$columnFilter.right` properties instead (provides avoid using [0] and [1] indexes in UI, but also required some additional logic to combine it to expression before submit).

Reply:
I'm not sure by switching operator type just change value type that will be enough, it's okey that value for different expression can handle different types. If expression will require some object structure for the value what we are going to do? 

I'm worring about this because it'll go to public and no chance to change this, because it's our API.